### DISCUSSION TITLE
Impressão em PDV no front-end

### DISCUSSION THREAD

**Discussion by nathpaiva** [frontendbr/forum#387] [Discussão]
Olá pessoal 😁 

Eu sou front-end e trabalho em uma statup e o nosso produto é uma praça de alimentação no celular, onde nós adiantamos o seu pedido e pagamento antes mesmo de estar na praça do shopping.

No front-end a gente tem um conector, que roda nas lojas, que é uma interface web e também um chrome app (em breve vamos migrar) e essa interface tem algumas funcionalidades, entre elas impressão em impressora térmica.

Hoje a solução que nós implementamos é utilizar o qz.io, é muito bacana e fácil de usar ele, porém requer algumas coisas para o funcionamento além da licença.

Alguém já trabalhou com esse tipo de coisa conhece alguma outra solução para fazer essa conexão da aplicação com o sistema?

Eu não conheço muito sobre impressão e impressoras e queria conversar sobre isso :)

**Comment by jonasporto** [comment]
Olá @nathpaiva, eu trabalho no [Gokoni.com](http://gokoni.com) e passei exatamente por esse problema. 

Aqui temos a plataforma de e-commerce e tínhamos a necessidade de imprimir o cupom, rota e outras rotinas nas impressoras térmicas.

A solução que chegamos aqui, resumidamente, foi criar uma fila de impressão(Web), e uma implementação de um aplicativo java(Desktop) simples que faz a rotina de buscar sua respectiva fila e executar a impressão.

**Comment by nathpaiva** [comment]
Bacana @jonasporto uma boa solução!

Melhor do que utilizar plataforma de terceiro. Estou em busca de uma melhoria para não depender do QZ e de atualizações, apesar dele ser muito bom é ruim ficar controlando versão de outras empresas.

Trabalha com impressão é um pouco chato!

Ainda mais que é difícil de achar pessoal desenvolvendo isso com front-end HIOHI se eu achar alguma coisa e mais alguma solução posto aqui. x)

**Comment by felipefialho** [comment]
Boa @nathpaiva 

Se resolver avisa nóis 👊

**Comment by nathpaiva** [comment]
Po certeza @LFeh, isso é bem chatinho se eu tiver uma resolução bacana com certeza compartilho com a galera! 

Que vai que alguém precisa fazer isso na vida hIOhiohIHOI :) 

Thanks Galera!!!

**Comment by vinibarr** [comment]
Olá; hoje é 29/04/2024; já surgiu alguma solução interessante para direcionamentos de impressora via front end ?
